#!/bin/sh /usr/share/dpatch/dpatch-run

## 17_replay.dpatch by Reinhard Nissl <rnissl@gmx.de>
## http://www.vr-web.de/~rnissl/vdr-1.3.11-replay.patch
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Replay patch fixes recording replay when reaching the end of a recording
## DP: in trickspeed mode. VDR switched to play mode instead of ending replay.

@DPATCH@
diff -urNad vdr-1.3.23/dvbplayer.c /tmp/dpep.OdxwwC/vdr-1.3.23/dvbplayer.c
--- vdr-1.3.23/dvbplayer.c	2005-04-18 22:46:41.002332578 +0100
+++ /tmp/dpep.OdxwwC/vdr-1.3.23/dvbplayer.c	2005-04-18 22:46:41.411299437 +0100
@@ -481,6 +481,12 @@
                              continue;
                           }
                        else {
+                          // hit end of recording: signal end of file but don't change playMode
+                          if (playDir == pdForward) {
+                             readIndex = -1;
+                             eof = true;
+                             continue;
+                             }
                           // hit begin of recording: wait for device buffers to drain
                           // before changing play mode:
                           if (!DeviceFlush(100))
